### PR TITLE
Add support for draft pull requests

### PR DESCRIPTION
**Pull request by JoeyBG**
This adds support for _draft pull requests_. Reviewers will be requested when the pull request is marked as  _Ready for Review_.

### CODE REVIEW

File: lib/dispatch_web/webhooks/controller.ex [hunk 1]
```diff
@@ -6,22 +6,26 @@ defmodule DispatchWeb.Webhooks.Controller do
...
 
       _ ->
         json(conn, %{success: true, noop: true})
     end
   end
 
-  def create(conn, _), do: json(conn, %{success: true, noop: true})
-
-  defp do_create(conn, params) do
+  defp assign_reviewers(conn, params) do
```
Comment (remi) [inline lib/dispatch_web/webhooks/controller.ex]: Is there a reason why you renamed this function?

Comment (JoeyBG) [reply lib/dispatch_web/webhooks/controller.ex]: I originally wanted to name `create_if_owner` `do_create` like we often do when we wish to re-use the result of arguments matching... I think I changed my mind and settled for a clearer name and forgot I had renamed the other one.

In any case, I can change it back, but I think the new name is slightly more explicit in what the function accomplishes.

Comment (remi) [reply lib/dispatch_web/webhooks/controller.ex]: The `do_<foo>` pattern is used in Elixir when validations cannot be ran as guard clauses. What we could do here is keep everything the way it was and just add this

```diff
-def create(conn, %{"action" => "opened", "repository" => %{"owner" => %{"login" => repo_owner}}} = params) do
+def create(conn, %{"action" => action, "repository" => %{"owner" => %{"login" => repo_owner}}} = params) when action in ~w(opened ready_for_review) do
```

There you go!

Comment (JoeyBG) [reply lib/dispatch_web/webhooks/controller.ex]: That's indeed quite simpler. Thanks!

**Review by Boubalou — APPROVED** [review]

**Review by remi — APPROVED** [review]
Awesome, good job! 👍

**Pull request merged**